src/SmartComponents/SystemAdvisories/SystemAdvisories.test.js

```diff
@@ -40,153 +41,67 @@ const mockState = {
 };
 
 const initStore = (state) => {
-    const customMiddleWare = () => next => action => {
-        useSelector.mockImplementation(callback => {
-            return callback({  SystemAdvisoryListStore: state });
-        });
-        next(action);
-    };
-
-    const mockStore = configureStore([customMiddleWare]);
+    const mockStore = configureStore([]);
     return mockStore({  SystemAdvisoryListStore: state });
 };
 
-let wrapper;
 let store = initStore(mockState);
-
 beforeEach(() => {
-    store.clearActions();
-    useSelector.mockImplementation(callback => {
-        return callback({ SystemAdvisoryListStore: mockState });
-    });
-    wrapper = mount(<Provider store={store}>
-        <Router><SystemAdvisories inventoryId='test' /></Router>
-    </Provider>);
-});
-
-afterEach(() => {
-    useSelector.mockClear();
+    render(<ComponentWithContext renderOptions={{ store }}>
+        <SystemAdvisories inventoryId='test' />
+    </ComponentWithContext>);
 });
 
 describe('SystemAdvisories.js', () => {
-    it('Should dispatch CHANGE_ADVISORY_LIST_PARAMS only once on load', () => {
-        const dispatchedActions = store.getActions();
-        expect(dispatchedActions.filter(item => item.type === 'CHANGE_SYSTEM_SYSTEMS_LIST_PARAMS')).toHaveLength(1);
-    });
-
-    it('Should dispatch expandAdvisoryRow action onCollapse', () => {
-        wrapper.find('TableView').props().onCollapse(null, 0, 'testValue');
-
-        const dispatchedActions = store.getActions();
-        expect(dispatchedActions.filter(item => item.type === 'EXPAND_SYSTEM_ADVISORY_ROW')).toHaveLength(1);
-        expect(dispatchedActions[1].payload).toEqual({
-            rowId: 'RHSA-2020:2774',
-            value: 'testValue'
-        });
-    });
-
-    it('Should collapse a row ', () => {
-        wrapper.find('TableView').props().onCollapse(null, 0, 'testValue');
+    it('should handle remediation', async() => {
+        const selectedState = {
+            ...mockState,
+            selectedRows: { 'RHSA-2020:2774': true }
+        };
 
-        const dispatchedActions = store.getActions();
-        expect(dispatchedActions.filter(item => item.type === 'EXPAND_SYSTEM_ADVISORY_ROW')).toHaveLength(1);
-        expect(dispatchedActions[1].payload).toEqual({
-            rowId: 'RHSA-2020:2774',
-            value: 'testValue'
+        const tempStore = initStore(selectedState);
+        render(
+            <ComponentWithContext renderOptions={{ store: tempStore }}>
+                <SystemAdvisories inventoryId='test' />
+            </ComponentWithContext>
+        );
+
+        waitFor(() => {
+            expect(AsyncRemediationButton).toHaveBeenCalledWith({
```

Comment:
Again, I feel that the title of the test is not aligned with what is actually tested